pkg/library/libraryinputresources/types.go

```diff
@@ -14,11 +14,21 @@ type InputResources struct {
 type ResourceList struct {
 	ExactResources []ExactResourceID `json:"exactResources,omitempty"`
 
+	// matchLabels defines a set of label key-value pairs for matching resources.
+	// Resources must have all specified labels to be included in the selection.
+	MatchLabels []MatchLabel `json:"matchLabels,omitempty"`
+
 	// use resourceReferences when one resource (apiserver.config.openshift.io/cluster) refers to another resource
 	// like a secret (.spec.servingCerts.namedCertificates[*].servingCertificates.name).
 	ResourceReference []ResourceReference `json:"resourceReferences,omitempty"`
 }
 
+// matchLabel represents a label selector in the form of a key-value pair.
+type MatchLabel struct {
```

Comment:
We probably need to limit this by resource and namespaces.  The resource is important because we have > 100 CRDs.  The namespace is important for permissioning purposes. We don't want to feed data from customer namespace or random extension into the the operator.